File: Binary Search/167_TwoSumII-Inputarrayissorted.py
class Solution(object):

    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        if len(numbers) < 2:
            return
        left, right = 0, len(numbers) - 1
        while left < right:
            des = numbers[left] + numbers[right]
            if des == target:
                return [left + 1, right + 1]
            elif des < target:
                left += 1
            else:
                right -= 1
        return

File: Binary Search/167_TwoSumII-Inputarrayissorted.java
public class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int[] result = {-1, -1};
        int n = numbers.length;
        if (n < 2) {
            return result;
        }
        int low = 0, high = n - 1;
        while (low < high - 1) {
            int mid = low + (high - low) / 2;
            if (target - numbers[mid] < numbers[mid]) {
                if (target == numbers[low] + numbers[mid]) {
                    result[0] = low;
                    result[1] = mid;
                    return result;
                }
                high = mid;
            } else if(target - numbers[mid] > numbers[mid]) {
                if (target == numbers[low] + numbers[mid]) {
                    result[0] = low;
                    result[1] = mid;
                    return result;
                }
                high = mid;
            }
        }
    }
}